slither/tools/read_storage/read_storage.py

```diff
@@ -240,16 +328,61 @@ def get_all_storage_variables(self, func: Callable = None) -> None:
             func (Callable, optional): A criteria to filter functions e.g. name.
         """
         for contract in self.contracts:
-            self._target_variables.extend(
-                filter(
-                    func,
-                    [
-                        (contract, var)
-                        for var in contract.state_variables_ordered
-                        if not var.is_constant and not var.is_immutable
-                    ],
-                )
-            )
+            for var in contract.state_variables_ordered:
+                if func is None or (func is not None and func(var)):
+                    if not var.is_constant and not var.is_immutable:
+                        self._target_variables.append((contract, var))
+                    elif var.is_constant and var.type == ElementaryType("bytes32"):
+                        self._constant_storage_slots.append((contract, var))
+            hardcoded_slot = self.find_hardcoded_slot_in_fallback(contract)
+            if hardcoded_slot is not None:
+                self._constant_storage_slots.append((contract, hardcoded_slot))
+
+    @staticmethod
+    def find_hardcoded_slot_in_fallback(contract: Contract) -> Optional[StateVariable]:
+        fallback = None
+        for func in contract.functions_entry_points:
+            if func.is_fallback:
+                fallback = func
+                break
+        if fallback is None:
+            return None
+        for node in fallback.all_nodes():
+            if node.type == NodeType.ASSEMBLY and isinstance(node.inline_asm, str):
+                asm_split = node.inline_asm.split("\n")
+                for asm in asm_split:
+                    if "sload(0x" in asm:  # Only handle literals
+                        arg = asm.split("sload(")[1].split(")")[0]
+                        exp = Literal(arg, ElementaryType("bytes32"))
+                        sv = StateVariable()
+                        sv.name = "fallback_sload_hardcoded"
+                        sv.expression = exp
+                        sv.is_constant = True
+                        sv.type = exp.type
+                        sv.set_contract(contract)
+                        return sv
+            elif node.type == NodeType.EXPRESSION:
+                exp = node.expression
+                if isinstance(exp, AssignmentOperation):
+                    exp = exp.expression_right
```

Comment:
This can be a nested expression. I wonder if this code could benefit from reusing `ConstantFolding` to account for nested expressions